scripts/build.sh

```diff
@@ -1,5 +1,6 @@
-#!/usr/bin/env bash
+#!/usr/bin/env nix-shell
+#! nix-shell --pure -i bash -I nixpkgs=/home/developer/nodejs/node/tools/nix/pkgs.nix /home/developer/nodejs/node/shell.nix
 
-set -e # Exit with nonzero exit code if anything fails 
+set -xe
 
-/home/developer/nodejs/node/configure --ninja && make -C /home/developer/nodejs/node
+make -C /home/developer/nodejs/node build-ci
```

Comment:
Does it have to use make? I think maybe it's about time that we can just use ninja in the GHA...it would be rather unfortunate to revert to make for development when the devcontainer is used, as it's not really pleasant to use for local development, and if the image is not built with ninja, then switching to ninja for development no longer hits the cache and everything has to be recompiled.

Reply:
The image is still using Ninja, it's defined in https://github.com/nodejs/node/blob/83ba6b107a067f5831b03ab69c787fcb35bdcb05/shell.nix#L87. The fact that the build is started with `make` predates this PR, the actual change in this PR is to use `build-ci` which calls `./configure` then `make all` (defined in https://github.com/nodejs/node/blob/83ba6b107a067f5831b03ab69c787fcb35bdcb05/Makefile#L627-L629).

FWIW that's also what test-shared is doing on nodejs/node, and as you can see in e.g. https://github.com/nodejs/node/actions/runs/20067281194/job/57559349798, we're using Ninja there.

Reply:
I see, thanks for the explanation - though I think to be safe this should be used:

```suggestion
BUILD_WITH=ninja make -C /home/developer/nodejs/node build-ci
```

(we should probably do it for the CI too - when `BUILD_WITH` is not set, the targets that depend on NODE_EXE etc. in the Makefile would try to build it again with Make IIUC - we don't currently have another NODE_EXE dependency in the build path, but if that happens it would be built twice)

Reply:
IMO it'd be preferable to not set it, and find another way to address your concern, let me explain. One of the upside of re-using Nix when building the Docker image is that we have basically one build env to maintain which covers both devcontainer users and Nix users. If the Dockerfile needs to deviate from the env provided by Nix, it kinda defeats the purpose, we should either include `BUILD_WITH` from the Nix env (although I don't think it's necessary), or not set it here.
wdyt about adding some assertion after the build instead? That way, we can still catch if for whatever reason `ninja` is not being used, and we can fix it for everyone on the Nix side.

Reply:
I am fine either way though I don't think adding assertions is useful - it would merely fail the build without fixing it, and it's uncertain whether anyone would be interested in fixing the double build issue. It would be nice if BUILD_WITH can be intelligently detected no matter what environment is used, though, but that's a bit out of scope for this PR.

Reply:
> we should either include BUILD_WITH from the Nix env (although I don't think it's necessary)

For actually working within the container (instead of only using it in the CI) I think this would be more helpful. There are a couple of times when I try to use other make targets when I build it with ninja and it would try to rebuild with make unless I explicitly set BUILD_WITH=ninja to pick up the binary built by ninja. IMO for actually using ninja for development, this env must be set anyway until the Makefile can be fixed to more intelligently use the ninja build for all targets, not just the ones used by the CI.

Reply:
Hum OK, I thought your concern were that the script no longer explicitely mentions `ninja`, which is why I suggested using an assertion. `BUILD_WITH` is already set by Nix; defined in https://github.com/nodejs/node/blob/83ba6b107a067f5831b03ab69c787fcb35bdcb05/shell.nix#L60.

Reply:
Oh I see, in that case this is resolved, though it's probably useful to leave a reference

```suggestion
# BUILD_WTIH=ninja is set in https://github.com/nodejs/node/blob/HEAD/shell.nix
make -C /home/developer/nodejs/node build-ci
```